enarx-keep-sgx-shim/src/start.S

```diff
@@ -142,6 +174,9 @@ enclave:
 .Lsyscall:
     movq    $0,                     SRSP(%r11)      # Clear syscall return stack pointer field
     mov     %rax,                   %rsp            # Restore the syscall return stack pointer
+    movq    $~0,                    %rax            # Set mask for xrstor in rax
+    movq    $~0,                    %rdx            # Set mask for xrstor in rdx
```

Comment:
You should probably only enable the supported bits from https://github.com/enarx/enarx/blob/master/sgx-types/src/attr.rs#L37 so that the size is not exceeded.

Reply:
@haraldh This is a good point. As far as I understand it, we can't read from `SECS`, so we can't access the current `xfrm` value. So I've done the next best thing - our default `xfrm` value sets bits `1:0` for `x87` and `SSE`, and no others. The new mask sets these same bits.